subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/PathUtil.java

```diff
@@ -74,31 +143,39 @@ static int compareChars(char char1, char char2) {
                 : -1
             : isFileSeparator(char2)
                 ? 1
-                : Character.compare(char1, char2);
+                : char1 - char2;
```

Comment:
I think it would be better to leave the original code here. It is going to be inlined anyway, yet the code reads much better that way.

Reply:
Using `Character.compare` again.